project/webapp/controller/ProductsList.controller.js

```diff
@@ -186,18 +190,68 @@ sap.ui.define(
               })
             );
           }
-          if (suppliersFilter) {
+          
             aFilters.push(
               new Filter({
                 filters: suppliersFilter,
                 and: false
               })
             );
-          }
+          
 
           return aFilters;
         },
-      }
-    );
+
+        onSearchProducts(oEvent) {
+          const oTableBinding = this.byId("productList").getBinding("items");
+          const aFilters = this._getAllFilters(oEvent);
+
+          oTableBinding.filter(aFilters);
+        },
+
+        onClearFilters() {
+          const oTableBinding = this.byId("productList").getBinding("items");
+
+          this.byId("releaseDate").setValue(null);
+          this.byId("supplier").setTokens([]);
+          this.byId("searchName").setValue(null);
+          this.byId("categorySelect").setSelectedKeys([]);
+
+          oTableBinding.filter(null);
+        },
+
+        _deleteProduct() {
+          // TODO: add products delete functionality
+        },
+
+        onDeleteProductPress() {
+          MessageBox.confirm(this._getConfirmationText(), {
+            title: this.oResourceBundle.getText("ConfirmDeleteProductTitle"),
+            actions: [MessageBox.Action.OK, MessageBox.Action.CLOSE],
+            onClose: (sAction) => {
+              switch (sAction) {
```

Comment:
we should use switch in case we need to handle more than 2-3 cases, in case you need to check only 1 condition, use 'if'